language/move-lang/src/expansion/ast.rs

```diff
@@ -51,6 +51,7 @@ pub type Attribute = Spanned<Attribute_>;
 pub struct Script {
     pub attributes: Vec<Attribute>,
     pub loc: Loc,
+    pub dependency_summary: BTreeSet<ModuleIdent>,
```

Comment:
What is this a summary of exactly? What are these idents?

If i read this correctly it is immediate dependencies? Maybe that would be a better name?

Reply:
It is a summary of immediate dependencies + immediate friends. Yes, I guess it needs a better name.

Reply:
Can we just make it a `BTreeMap<ModuleIdent, UseOrFriend>` or something? I and just call it `immediate_neighbors`

I get that it is more information than you need in the prover but otherwise it feels super confusing to me

Reply:
Thanks  for the suggestion. That should work.

Reply:
I eventually made an `immediate_neighbors: BTreeSet<Neighbor>` where

```rust
pub enum Neighbor {
    Dependency(ModuleIdent),
    Friend(ModuleIdent),
}
```

The reason I did not go for `BTreeMap<ModuleIdent, UseOrFriend>` is that in a malformed program, there could be a `ModuleIdent` mapping to both `Use` and `Friend`, and having a `BTreeMap<ModuleIdent, BTreeSet<UseOrFriend>>` looks a bit cumbersome and does not capture more information than the `BTreeSet<Neighbor>`.